src/Service/Configurations/RuntimeConfigValidator.cs

```diff
@@ -239,25 +239,70 @@ public static void ValidateEntitiesDoNotGenerateDuplicateQueriesOrMutation(IDict
                 {
                     throw new DataApiBuilderException(
                         message: $"Entity {entityName} generates queries/mutation that already exist",
-                        statusCode: System.Net.HttpStatusCode.ServiceUnavailable,
+                        statusCode: HttpStatusCode.ServiceUnavailable,
                         subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError);
                 }
             }
         }
 
         /// <summary>
-        /// Check whether the entity name defined in runtime config only contains
-        /// characters allowed for GraphQL names.
-        /// Does not perform validation for entities which do not
+        /// Check whether the entity configuration defined in runtime config only contains characters allowed for GraphQL names
+        /// and other validations related to rest path and methods configured for the entity.
+        /// The GraphQL validation is not performed for entities which do not
         /// have GraphQL configuration: when entity.GraphQL == false or null.
         /// </summary>
         /// <seealso cref="https://spec.graphql.org/October2021/#Name"/>
-        /// <param name="runtimeConfig"></param>
-        public static void ValidateEntityNamesInConfig(Dictionary<string, Entity> entityCollection)
+        /// <param name="runtimeConfig">The runtime configuration.</param>
+        public static void ValidateEntityConfiguration(RuntimeConfig runtimeConfig)
         {
-            foreach (string entityName in entityCollection.Keys)
+            // Stores the unique rest paths configured for different entities present in the config.
+            HashSet<string> restPathsForEntities = new();
+
+            foreach (KeyValuePair<string, Entity> Entity in runtimeConfig.Entities)
```

Comment:
nit: calling the pair as Entity seems inappropriate since the value is the Entity. Also, the name of this pair variable should be camelCased since its a function local variable.